debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerChangeEventSourceCoordinator.java

```diff
@@ -0,0 +1,178 @@
+                            schema.assureNonEmptySchema();
+                        }
+
+                        if (snapshotResult.isCompletedOrSkipped()) {
+                            streamingOffsets.getOffsets().put(partition, snapshotResult.getOffset());
+                        }
+                    }
+
+                    streamEvents(previousLogContext, taskContext, context, streamingOffsets);
+                }
+                catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                    LOGGER.warn("Change event source executor was interrupted", e);
+                }
+                catch (Throwable e) {
+                    errorHandler.setProducerThrowable(e);
+                }
+                finally {
+                    streamingConnected(false);
+                }
+            });
+        }
+        finally {
+            if (previousLogContext.get() != null) {
+                previousLogContext.get().restore();
+            }
+        }
+    }
+
+    protected void streamEvents(AtomicReference<LoggingContext.PreviousContext> previousLogContext, CdcSourceTaskContext taskContext, ChangeEventSourceContext context,
+                                Offsets<SqlServerPartition, SqlServerOffsetContext> streamingOffsets)
+            throws InterruptedException {
+        final Metronome metronome = Metronome.sleeper(pollInterval, clock);
+
+        LOGGER.info("Starting streaming");
+
+        streamingSource = changeEventSourceFactory.getStreamingChangeEventSource();
+        eventDispatcher.setEventListener(streamingMetrics);
+        streamingConnected(true);
+        streamingSource.init();
+
+        while (running) {
+            boolean streamedEvents = false;
+            for (Map.Entry<SqlServerPartition, SqlServerOffsetContext> entry : streamingOffsets) {
+                SqlServerPartition partition = entry.getKey();
+                SqlServerOffsetContext previousOffset = entry.getValue();
+
+                if (running) {
+                    previousLogContext.set(taskContext.configureLoggingContext("streaming"));
+                    streamEvents(context, partition, previousOffset);
+                    if (((SqlServerStreamingChangeEventSource) streamingSource).eventsStreamed(partition)) {
+                        streamedEvents = true;
+                    }
+                }
+            }
+
+            if (!streamedEvents) {
+                metronome.pause();
+            }
+        }
+
+        LOGGER.info("Finished streaming");
+    }
+
+    @Override
+    protected void streamEvents(ChangeEventSource.ChangeEventSourceContext context, SqlServerPartition partition,
+                                SqlServerOffsetContext offsetContext)
+            throws InterruptedException {
+        // TODO: Determine how to do incremental snapshots with multiple partitions.
```

Comment:
Do we even have to override this method at all currently?

Reply:
It's been refactored for the deduplication and the override is now gone.